### ISSUE TITLE
"Step 1: Install Video2X and Dependencies" fails on pdm.pep517.exceptions.PEP621ValidationError: {'license-expression': ['unknown field']}

### ISSUE THREAD

**Issue by Almeros** [k4yt3x/video2x#755]
When running Step 1 in the Colab I get the following error. As far as I know I followed step 0 correctly. I tried it several times yesterday and today. Restarting the runtime many times. I'm not familiar with Colab so perhaps the problem just lies with me not understanding what to do?

> ...
> WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
>   Installing build dependencies ... done
>   Running command Getting requirements to build wheel
>   Traceback (most recent call last):
>     File "/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
>       main()
>     File "/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
>       json_out['return_val'] = hook(**hook_input['kwargs'])
>     File "/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
>       return hook(config_settings)
>     File "/tmp/pip-build-env-5y2y85sv/overlay/lib/python3.7/site-packages/pdm/pep517/api.py", line 26, in get_requires_for_build_wheel
>       if builder.meta.config.run_setuptools:
>     File "/tmp/pip-build-env-5y2y85sv/overlay/lib/python3.7/site-packages/pdm/pep517/base.py", line 154, in meta
>       self._meta.validate(True)
>     File "/tmp/pip-build-env-5y2y85sv/overlay/lib/python3.7/site-packages/pdm/pep517/metadata.py", line 58, in validate
>       return validate_pep621(self.data, raising)
>     File "/tmp/pip-build-env-5y2y85sv/overlay/lib/python3.7/site-packages/pdm/pep517/validator.py", line 85, in validate_pep621
>       raise PEP621ValidationError(validator.errors)
>   pdm.pep517.exceptions.PEP621ValidationError: {'license-expression': ['unknown field']}
>   error: subprocess-exited-with-error
>   
>   × Getting requirements to build wheel did not run successfully.
>   │ exit code: 1
>   ╰─> See above for output.
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   full command: /usr/bin/python3.7 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpnrv61vuo
>   cwd: /tmp/pip-install-kzlt0tky/video2x_c67fbdfff34a4f63a53bc6fa4a542a55
>   Getting requirements to build wheel ... error
> error: subprocess-exited-with-error
> 
> × Getting requirements to build wheel did not run successfully.
> │ exit code: 1
> ╰─> See above for output.
> 
> note: This error originates from a subprocess, and is likely not a problem with pip.

----

Edit: this happens in 5.0.0 beta-6 but works in 5.0.0 beta-4, so if you also encounter this problem, you can change it in colab in the script of Step 1. 

For me this works, but the bug/problem is still there in beta-6 so I'm not sure that this issue should be closed.

**Comment by jedidiahgamaliel** [comment]
Ran into the same problem.

**Comment by rogerb831** [comment]
I'm also getting this error.

**Comment by stavlocker** [comment]
Can confirm this doesn't happen in 5.0.0 beta-4, so you could try changing that in the installation script (even though I am currently encountering different issues after doing this).

**Comment by quito96** [comment]
Same problem here

**Comment by ColtonWehking** [comment]
> Can confirm this doesn't happen in 5.0.0 beta-4, so you could try changing that in the installation script (even though I am currently encountering different issues after doing this).

I can confirm beta-4 is working fine. As far as other issues, I encountered many too. Eventually I upgraded to a paid google colab, $10, and everything got fixed. Might be some resource issue?

**Comment by PlutoHDDev** [comment]
I am facing the same problem.

**Comment by quito96** [comment]
> I am facing the same problem.

switch to  5.0.0 beta-4 this works ;)

**Comment by xmusjackson** [comment]
It's an issue with the pyproject.toml. PDM complains about something in the license field, but I haven't tried to figure out why yet.
Just comment out the 'license-expression' field and all will work.

diff --git a/a.toml b/b.toml
index 6edaf1c..c54c2e9 100644
--- a/a.toml
+++ b/b.toml
@@ -3,7 +3,7 @@ name = "video2x"
 description = "A video/image upscaling and frame interpolation framework"
 readme = "README.md"
 requires-python = ">=3.7"
-#license-expression = "AGPL-3.0-or-later"
+license-expression = "AGPL-3.0-or-later"
 keywords = [
   "super-resolution",
   "upscaling",

**Comment by xmusjackson** [comment]
It's an issue with the pyproject.toml. Based on this commit to pdm-pep517 (the backend used for this project):
https://github.com/pdm-project/pdm-pep517/commit/aa7aecb8815740d7b6090e1ecfc0174b850330c0,
'license-expression' is once again 'license'. The PEP that defines the field as 'license-expression' is apparently out of date and the field name has been reverted from what I can tell. Either way, it's easy to work around. Just change the 'license-expression' field to just 'license' in pyproject.toml and all will work.

diff --git a/pyproject.toml b/pyproject.toml
index 6edaf1c..c54c2e9 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,7 +3,7 @@ name = "video2x"
description = "A video/image upscaling and frame interpolation framework"
readme = "README.md"
requires-python = ">=3.7"
-license-expression = "AGPL-3.0-or-later"
+license = "AGPL-3.0-or-later"
keywords = [
"super-resolution",
"upscaling",

I submitted a pull request with this patch.
